plugins/synapticsmst/fu-synapticsmst-device.c

```diff
@@ -1095,24 +1077,26 @@ synapticsmst_device_write_firmware (SynapticsMSTDevice *self,
 	const guint8 *payload_data;
 	gsize payload_len;
 	guint16 tmp;
-	SynapticsMSTChipKind chip_type = SYNAPTICSMST_CHIP_KIND_UNKNOWN;
+	FuSynapticsmstFamily chip_type = FU_SYNAPTICSMST_FAMILY_UNKNOWN;
 	g_autoptr(FuDeviceLocker) locker = NULL;
 
 	payload_data = g_bytes_get_data (fw, &payload_len);
 
-	if (synapticsmst_device_get_chip_id (self) > 0x5000)
-		chip_type = SYNAPTICSMST_CHIP_KIND_PANAMERA;
-	else
-		chip_type = SYNAPTICSMST_CHIP_KIND_TESLA_LEAF;
+	if (fu_synapticsmst_device_get_chip_id (self) > 0x5000)
+		chip_type = FU_SYNAPTICSMST_FAMILY_PANAMERA;
+	else if (fu_synapticsmst_device_get_chip_id (self) > 0x3000)
+		chip_type = FU_SYNAPTICSMST_FAMILY_LEAF;
+	else if (fu_synapticsmst_device_get_chip_id (self) > 0x2000)
+		chip_type = FU_SYNAPTICSMST_FAMILY_TESLA;
```

Comment:
I would think these 3 ifs should be inverted, since presumably based on the pattern whatever comes after panamera will be bigger and you don't need to change the first if statement.

Reply:
I wanted it to "degrade" so that anything much bigger that panamera would be tagged as panamera, rather than tesla. Or do I misunderstand?

Reply:
I do agree something after panamera is more likely to look like panamera.

However given the types of changes that were made with panamera I wouldn't be surprised if the next chip requires significant changes again. I do think it's safer to detect the next chip as unknown and not let update until code changes are made for it.

Reply:
Agreed. I've added this:

    FuSynapticsmstFamily
    fu_synapticsmst_family_from_chip_id (guint16 chip_id)
    {
        if (chip_id >= 0x5000 && chip_id < 0x6000)
            return FU_SYNAPTICSMST_FAMILY_PANAMERA;
        if (chip_id >= 0x3000 && chip_id < 0x4000)
            return FU_SYNAPTICSMST_FAMILY_LEAF;
        if (chip_id >= 0x2000 && chip_id < 0x3000)
            return FU_SYNAPTICSMST_FAMILY_TESLA;
        return FU_SYNAPTICSMST_FAMILY_UNKNOWN;
    }